src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetHash.cs

```
@@ -0,0 +1,349 @@
+        /// This is for paths collecting from pipe
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            List<string> pathsToProcess = new List<string>();
+            ProviderInfo provider = null;
+
+            switch (ParameterSetName)
+            {
+                case PathParameterSet:
+                    // Resolve paths and check existence
+                    foreach (string path in _paths)
+                    {
+                        try
+                        {
+                            Collection<string> newPaths = Context.SessionState.Path.GetResolvedProviderPathFromPSPath(path, out provider);
+                            if (newPaths != null)
+                            {
+                                pathsToProcess.AddRange(newPaths);
+                            }
+                        }
+                        catch (ItemNotFoundException e)
+                        {
+                            if (!WildcardPattern.ContainsWildcardCharacters(path))
+                            {
+                                ErrorRecord errorRecord = new ErrorRecord(e,
+                                    "FileNotFound",
+                                    ErrorCategory.ObjectNotFound,
+                                    path);
+                                WriteError(errorRecord);
+                            }
+                        }
+                    }
+                    break;
+                case LiteralPathParameterSet:
+                    foreach (string path in _paths)
+                    {
+                        string newPath = Context.SessionState.Path.GetUnresolvedProviderPathFromPSPath(path);
+
+                        // Unlike 'GetResolvedProviderPathFromPSPath'
+                        // 'GetUnresolvedProviderPathFromPSPath' does not check a file existence
+                        // so do that explicity
+                        if (File.Exists(newPath))
```

Also - ignoring exceptions is usually a terrible idea - I think non-terminating errors is definitely preferred.